examples/serialization/avro/avro-cli.py

```diff
@@ -0,0 +1,180 @@
+
+    def producer_conf(self):
+        return self._build()
+
+    def consumer_conf(self):
+        return dict({"group.id": self.group}, **self._build())
+
+
+class User(dict):
+    _schema = avro.loads(record_schema_str)
+
+    def __init__(self, conf=None):
+        if not conf:
+            conf = {"name": "anonymous",
+                    "favorite_number": 0,
+                    "favorite_color": ""}
+        super(User, self).__init__(conf)
+
+    @classmethod
+    def schema(cls):
+        return cls._schema
+
+    def prompt(self):
+        self['name'] = input("Enter name:")
+        num = input("Enter favorite number:")
+        if num == '':
+            num = 0
+        self['favorite_number'] = int(num)
+        self['favorite_color'] = input("Enter favorite color:")
+
+
+def on_delivery(err, msg):
+    if err is not None:
+        print('Message delivery failed ({} [{}]): %{}'.format(msg.topic(), str(msg.partition()), err))
+        return 0
+    elif err is not None:
+        print('Message delivery failed {}'.format(err))
+        return 0
+    else:
+        print('Message delivered to {} [{}] at offset [{}]: {}'.format(
+              msg.topic(), msg.partition(), msg.offset(), msg.value()))
+        return 1
+
+
+def produce(args):
+    from confluent_kafka.avro import AvroProducer
+
+    record = User()
+    topic = args.topic
+    conf = args.producer_conf()
+
+    producer = AvroProducer(conf, default_value_schema=record.schema())
+
+    print("producing user records to topic {}. ^c to exit.".format(topic))
+
+    while True:
+        try:
+            record.prompt()
+            producer.produce(topic=topic, partition=0, value=record, callback=on_delivery)
+            producer.poll(0.1)
+        except KeyboardInterrupt:
+            break
+        except ValueError:
+            print("Invalid input,  discarding record...")
+            continue
+        record = User()
+
+    print("\nFlushing records...")
+    producer.flush()
+
+
+def consume(args):
+    from confluent_kafka.avro import AvroConsumer
+    from confluent_kafka import KafkaError
+    from confluent_kafka.avro.serializer import SerializerError
+
+    topic = args.topic
+    conf = args.consumer_conf()
+    print("consuming user records from topic {}".format(topic))
+
+    c = AvroConsumer(conf)
+    c.subscribe([topic])
+
+    while True:
+        try:
+            msg = c.poll(1)
+
+        except SerializerError as e:
+            print("Message deserialization failed {}".format(e))
+            break
+
+        if msg is None:
+            continue
+
+        if msg.error():
+            if msg.error().code() == KafkaError._PARTITION_EOF:
+                break
+            else:
+                print(msg.error())
+                break
+
+        record = User(msg.value())
+        print("\nusername: {}\n\tfavorite_number:{}\n\tfavorite_color:{}".format(
+              record["name"], record["favorite_number"], record["favorite_color"]))
```

Comment:
Using a dict-type for User sort of defeats the purpose of having an explicit type, since dict can represent any complex type.
Suggest using a custom class with specific fields for name, favourite_nu.. and so on